Marlin/src/gcode/calibrate/M100.cpp

```diff
@@ -60,16 +60,61 @@
 
 #define TEST_BYTE ((char) 0xE5)
 
-extern char* __brkval;
-extern size_t  __heap_start, __heap_end, __flp;
-extern char __bss_end;
+#if defined(__AVR__) || IS_32BIT_TEENSY
+
+  extern char __bss_end;
+  char* end_bss = (
+    #ifdef __AVR__
+      int
+    #else
+      char*
+    #endif
+  )&__bss_end;
+  char* free_memory_start = end_bss;
```

Comment:
@Bob-the-Kuhn — It seems like just `char* end_bss = &__bss_end` should work for all cases. I'm not sure why __AVR__ had the `int` cast.

Reply:
> I'm not sure why AVR had the int cast.

Because AVR's RAM memory space was only 8KB.    An 'int' will hold that nicely.    
And that played into the routines that could output values.    (We still don't have routines
to output pointer values nicely.)

Moving forward, everything should be using 'void *ptr'.

Reply:
Just getting rid of the (int) makes bioth AVR and Teensy happy.

Reply:
Just to Nit-Pik....   Teensy *IS* an AVR part.

Reply:
I've pushed the simpler fix.

Scott - I have a copy of the file you created.  I'll put that one back on the PR if you want.